x-pack/plugins/security_solution/docs/prebuilt_rules_customization_rfc.md

```diff
@@ -0,0 +1,2867 @@
+  description: Discriminated union that determines whether the rule is internally sourced (created within the Kibana app) or has an external source, such as the Elastic Prebuilt rules repo.
+  oneOf:
+    - $ref: '#/components/schemas/ExternalRuleSource'
+    - $ref: '#/components/schemas/InternalRuleSource'
+#  [... file continues below ...]
+```
+
+_Source: [x-pack/plugins/security_solution/common/api/detection_engine/model/rule_schema/rule_schemas.schema.yaml](https://github.com/elastic/kibana/blob/main/x-pack/plugins/security_solution/common/api/detection_engine/model/rule_schema/rule_schemas.schema.yaml)_
+
+```yaml
+#  [... file continues above...]
+
+ResponseFields:
+  type: object
+  properties:
+    id:
+      $ref: './common_attributes.schema.yaml#/components/schemas/RuleObjectId'
+    rule_id:
+      $ref: './common_attributes.schema.yaml#/components/schemas/RuleSignatureId'
+    immutable:
+      $ref: './common_attributes.schema.yaml#/components/schemas/IsRuleImmutable'
+    rule_source:
+      $ref: './common_attributes.schema.yaml#/components/schemas/RuleSource'
+    #  [...  more response fields ...]
+  required:
+    - id
+    - rule_id
+    - immutable
+    - updated_at
+    - updated_by
+    - created_at
+    - created_by
+    - revision
+    - rule_source
+#  [... file continues below...]
+```
+
+##### Rule Import request schema
+
+We also need to modify the `RuleToImport` schema, since now we will be allowing the importing of both custom rules and prebuilt rules.
+
+Currently, `RuleToImport` optionally accepts the `immutable` param, but rejects validation if its value is set to anything else than `false` - since we don't currently support importing prebuilt rules. 
+
+We will be changing the mechanism for importing rules so that, besides the `rule_id`, the `version` of the rule is also a required parameter. These two parameters will be used to determine if the rule is prebuilt or not, and dynamically calculate `rule_source` during import.
+
+See the detailed explanation for this mechanism in the [Exporting and importing rules](#exporting-and-importing-rules) sections.
+
+The `immutable` and `rule_source` fields will be ignored if passed in the request payload.
+
+_Source: [x-pack/plugins/security_solution/common/api/detection_engine/rule_management/import_rules/rule_to_import.ts](https://github.com/elastic/kibana/blob/main/x-pack/plugins/security_solution/common/api/detection_engine/rule_management/import_rules/rule_to_import.ts)_
+
+```ts
+// [... file continues above...]
+import {
+  // [...]
+  RuleSignatureId,
+  IsRuleImmutable,
+  ExternalSourceAttributes,
+  RuleVersion,
+} from '../../model/rule_schema';
+
+export type RuleToImport = z.infer<typeof RuleToImport>;
+export type RuleToImportInput = z.input<typeof RuleToImport>;
+export const RuleToImport = BaseCreateProps.and(TypeSpecificCreateProps).and(
+  ResponseFields.partial().extend({
+    rule_id: RuleSignatureId,
+    version: RuleVersion,
+  })
+);
```

Comment:
`ResponseFields` is going to include both `immutable: IsRuleImmutable` and `rule_source: RuleSource.optional()`, according to changes in this PR. These two fields should NOT be included in the `RuleToImport` schema. This type definition will need to be adjusted somehow.

Reply:
@jpdjere You are right, from that perspective it should be fine to keep these fields in the `RuleToImport` schema and just ignore the input and recalculate them inside the import endpoint.